src/app/components/normal-component/demo-page/demo-page.component.ts

```diff
@@ -65,22 +66,19 @@ export class DemoPageComponent {
             const solution: Move | string = step.getSolution();
             if (typeof solution === 'string') {
                 return {
-                    node: new MGPNode(step.state),
+                    node: new GameNode(step.state),
                     click: MGPOptional.of(solution),
                 };
             } else {
                 const move: Move = solution;
+                const state: GameState = rules.applyLegalMove(move, step.state, rules.isLegal(move, step.state).get());
                 const node: AbstractNode =
-                    new MGPNode(rules.applyLegalMove(move,
-                                                     step.state,
-                                                     rules.isLegal(move, step.state).get()),
-                                MGPOptional.of(new MGPNode(step.state)),
-                                MGPOptional.of(move));
+                    new GameNode(state, MGPOptional.of(new GameNode(step.state)), MGPOptional.of(move));
```

Comment:
const legalityStatus: L = rules.isLegal(move, step.state).get();
const state: GameState = rules.applyLegalMove(move, step.state, legalityStatus);
const motherNode: AbstractNode = MGPOptional.of(new GameNode(step.state));
const node: AbstractNode = new GameNode(state, motherNode, MGPOptional.of(move));

C'est pas 700 fois mieux :D ?

Reply:
c'est... le code que j'ai devant les yeux déjà :D Donc j'suis pas sûr de ce qui c'est passé, mais je dirais "outdated" !

Reply:
Ah c'est pas outdaté du tout et c'est pas le code que je vois :D !

Reply:
benwai j'lis comme d'la marde, corrigé !